### PR TITLE
[Synapse]update to  AZ Synapse mgmt 2.0.0

### PR DESCRIPTION
**Pull request by zesluo**
**Description**
update to AZ Synapse mgmt 2.0.0

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Synapse

**Comment by jsntcy** [comment]
@zesluo, please fix failed tests first.

File: src/azure-cli/azure/cli/command_modules/synapse/tests/latest/test_synapse_scenario.py [hunk 1]
```diff
@@ -318,22 +342,25 @@ def test_sql_pool_restore_and_list_deleted(self):
...
+        time.sleep(200)
+
         # test list-deleted
         self.cmd('az synapse sql pool list-deleted --workspace-name {workspace} --resource-group {rg}',
                  checks=[
                      self.greater_than("length([])", 0)
                  ])
 
-    @record_only()
+    #@record_only()
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/synapse/tests/latest/test_synapse_scenario.py:354]: Is it expected that we commented out `record_only`?

File: src/azure-cli/azure/cli/command_modules/synapse/tests/latest/recordings/test_sql_pool_classification_and_recommendation.yaml [hunk 2]
```diff
@@ -14,15 +14,15 @@ interactions:
...
       User-Agent:
-      - python/3.7.2 (Windows-10-10.0.18362-SP0) msrest/0.6.18 msrest_azure/0.6.3
-        azure-mgmt-security/0.4.1 Azure-SDK-For-Python AZURECLI/2.16.0
+      - python/3.7.4 (Windows-10-10.0.19041-SP0) msrest/0.6.21 msrest_azure/0.6.3
+        azure-mgmt-security/0.6.0 Azure-SDK-For-Python AZURECLI/2.24.0
       accept-language:
       - en-US
     method: GET
-    uri: https://management.azure.com/providers/Microsoft.Management/managementGroups/00000000-0000-0000-0000-000000000000/providers/Microsoft.Security/informationProtectionPolicies/effective?api-version=2017-08-01-preview
+    uri: https://management.azure.com/providers/Microsoft.Management/managementGroups/72f988bf-86f1-41af-91ab-2d7cd011db47/providers/Microsoft.Security/informationProtectionPolicies/effective?api-version=2017-08-01-preview
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/synapse/tests/latest/recordings/test_sql_pool_classification_and_recommendation.yaml:22]: It seems the recording file is wrong as usually it should be replaced by mocked value.

**Review by jsntcy — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/synapse/_params.py [hunk 3]
```diff
@@ -232,10 +232,13 @@ def load_arguments(self, _):
         c.argument('sql_pool_name', arg_type=name_type, id_part='child_name_1', help='The SQL pool name.')
         c.argument('status', arg_type=get_enum_type(TransparentDataEncryptionStatus),
                    required=True, help='Status of the transparent data encryption.')
+        c.argument('transparent_data_encryption_name', options_list=['--trans-data-encry', '-d'], help='Name of the transparent data encryption.')
 
     # synapse sql pool threat-policy
     with self.argument_context('synapse sql pool threat-policy') as c:
         c.argument('sql_pool_name', arg_type=name_type, id_part='child_name_1', help='The SQL pool name.')
+        c.argument('security_alert_policy_name', options_list=['--sec-alert-ply', '-s'],
```
Comment (idear1203) [inline src/azure-cli/azure/cli/command_modules/synapse/_params.py:240]: Can we use abbreviation for argument names in CLI? @jsntcy

**Review by idear1203 — COMMENTED** [review]
I found several abbreviation occurrences. Is this allowed? @jsntcy

File: src/azure-cli/azure/cli/command_modules/synapse/_params.py (hunk 3 above)
Comment (jsntcy) [reply src/azure-cli/azure/cli/command_modules/synapse/_params.py]: @idear1203, what do you mean? `options_list=['--sec-alert-ply', '-s']` is the abbreviation.

Comment (zesluo) [reply src/azure-cli/azure/cli/command_modules/synapse/_params.py]: fixed

**Review by idear1203 — APPROVED** [review]
LGTM.

File: src/azure-cli/azure/cli/command_modules/synapse/_params.py (hunk 3 above)
Comment (idear1203) [reply src/azure-cli/azure/cli/command_modules/synapse/_params.py]: @jsntcy sorry for the confusion. Do we allow `--sec-alert-ply` as argument name? Previously I always use full words like `--security-alert-policy`

**Pull request merged**